.github/workflows/multichain-prod.yml

```diff
@@ -0,0 +1,58 @@
+name: Deploy Multichain Prod.
+on:
+  workflow_dispatch:
+    inputs:
+      network:
+        type: choice
+        options:
+          - mainnet
+          - testnet
+        description: mainnet or testnet network
+        required: true
+      image:
+        description: Full Artifact Registry image with tag (e.g. us-east1-docker.pkg.dev/pagoda-discovery-platform-prod/multichain/multichain-< testnet | mainnet >)
+        required: true
+      tag:
+        description: Image tag that you wish to deploy, either by SHA or Version/latest
+
+jobs:
+  build-mpc-recovery:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        name: "Checkout mpc-recovery"
+
+      - name: Login to Artifact Registry
+        uses: docker/login-action@v2
+        with:
+          registry: us-east1-docker.pkg.dev
+          username: _json_key
+          password: ${{ secrets.GCP_CREDENTIALS_PROD }}
+
+      - name: Build Docker image and push to Google Artifact Registry
+        id: docker-push-tagged
+        uses: docker/build-push-action@v4
+        with:
+          push: true
+          file: ./Dockerfile.multichain
+          tags: "${{ github.event.inputs.image }}:${{ github.event.inputs.tag }}"
+
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - id: 'auth'
+        uses: 'google-github-actions/auth@v2'
+        with:
+          credentials_json: '${{ secrets.GCP_CREDENTIALS_PROD }}'
+  
+      - name: 'Set up Cloud SDK'
+        uses: 'google-github-actions/setup-gcloud@v2'
+  
+      - name: 'Set project'
+        run: 'gcloud config set project pagoda-discovery-platform-prod'
+
+      - name: 'Update Nodes'
+        run: |
+          gcloud compute instances update-container multichain-${{ github.event.inputs.network }}-0
+          gcloud compute instances update-container multichain-${{ github.event.inputs.network }}-1
+          gcloud compute instances update-container multichain-${{ github.event.inputs.network }}-2
```

Comment:
Let's fix this, there is no newline at the end of half the files.

Reply:
done